### PR TITLE
PCHR-2446: Default sick leave to single day

### PR DESCRIPTION
**Pull request by igorpavlov**
## Overview

This PR defaults the sickness request recording to a single day mode.

## Technical Details

1. A new private function `setDaySelectionMode` was added, it includes the old code that set multiple/single day UI option as well as the new code that checks for leave request type (sick) and mode (create).
2. This function is now called earlier that the original code was - before the selection/pre-selection of the contact so the UI is correct at all times.

## Comments

1. Along with the solution a quick UX improvement is applied. The Record Absence dropdown items did not have a pointer cursor on them. Empty "href" attribute is added to items.
2. ✨ Request controller and it's children are partially refactored according to the Style Guide.
  - Previously non-private functions are now private and bound to controller separately
  - All functions have been sorted alphabetically (functions were NOT renamed, underscored functions remained underscored)
  - Comments added to some functions
  - Init functions renamed

_Note: Some files in the refactoring commit https://github.com/civicrm/civihr/pull/2042/commits/03f1dcb51da3824acde568a47cdfd53f2c8511a
  contain to many changes, so diffs are hidden by default. This is mostly due to the function sorting._

---

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/sick-request.controller.js [hunk 1]
```diff
@@ -22,26 +22,34 @@ define([
         }
       };
 
+      vm.canSubmit = canSubmit;
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/sick-request.controller.js]: Use `$inject` for all controllers/components touched here.
https://github.com/civicrm/civihr/blob/8fe49c2c7ab52d1d3da0f95dc4c397a0d7ca26f6/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-notification-badge.component.js#L18

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/sick-request.controller.js [hunk 2]
```diff
@@ -109,6 +102,15 @@ define([
...
 
+      /**
+       * Initialize leaverequest based on attributes that come from directive
+       */
+      function _initRequest () {
+        var attributes = vm._initRequestAttributes();
+
+        vm.request = SicknessRequestInstance.init(attributes);
+      }
+
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/sick-request.controller.js]: `return` statement not required anymore.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request.controller.js [hunk 3]
```diff
@@ -291,6 +258,49 @@ define([
...
+        var attributes = vm._initRequestAttributes();
+
+        vm.request = TOILRequestInstance.init(attributes);
+        /*
+         * toil request does not have date type but leave request requires it
+         * for validation, hence setting it to All Day's value which is 1
+         */
+        vm.request.to_date_type = vm.request.from_date_type = '1';
+      }
+
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request.controller.js]: return statement not necessary

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/sick-request.controller.js (hunk 2 above)
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/sick-request.controller.js]: It is a child controller, you must return.

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request.controller.js (hunk 3 above)
Comment (igorpavlov) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request.controller.js]: Same as above.

**Pull request merged**